app/src/main/java/org/nutritionfacts/dailydozen/model/pref/UpdateReminderPref.java

```diff
@@ -148,6 +148,14 @@ private int timeInMillis(String time, DateFormat format) {
                     return (int) date.getTime();
                 }
             } catch (ParseException e) {
+                // If Canada Locale, AM/PM needs to be replaced with a.m./p.m. otherwise the time will not parse
+                // So if we get a ParseException, try replacing AM/PM with a.m./p.m. and try again
+                if (time.contains("AM") || time.contains("PM")) {
+                    time = time.replace("AM", "a.m.").replace("PM", "p.m.");
+                    return timeInMillis(time, format);
+                }
```

Comment:
The solution works. However, I was wondering if it would be better to directly define `twelveHourFormat` with `Locale.US` instead of `Locale.getDefault()`, as we know it will always come similar to "8:00 PM". That way we wouldn't need a specific workaround for Canada locale and could potentially fix issues with other locales.

Reply:
I went with your suggestion, but I'm doing it in another branch as I have other updates: https://github.com/nutritionfactsorg/daily-dozen-android/pull/235